src/Raven.Server/Documents/DatabasesLandlord.cs

```diff
@@ -168,15 +168,62 @@ private void HandleClusterDatabaseChanged(string databaseName, long index, strin
             }
         }
 
-        private void UnloadDatabase(string databaseName)
+        private void UnloadDatabase(string databaseName, bool dbRecordIsNull = false)
+        {
+            if (dbRecordIsNull)
+            {
+                UnloadDatabaseInternal(databaseName);
+                return;
+            }
+
+            if (DatabasesCache.TryGetValue(databaseName, out var databaseTask) == false)
+            {
+                // database was already unloaded or deleted
+                return;
+            }
+
+            if (databaseTask.IsCompletedSuccessfully)
+            {
+                UnloadDatabaseInternal(databaseName);
+            }
+            else if (databaseTask.IsCompleted == false)
+            {
+                // This case is when an unload was issued prior to the actual loading of a database.
+                databaseTask.ContinueWith(t =>
+                {
+                    if (databaseTask.IsCompletedSuccessfully)
+                    {
+                        using (_serverStore.ContextPool.AllocateOperationContext(out TransactionOperationContext context))
+                        using (context.OpenReadTransaction())
+                        {
+                            var databaseRecord = _serverStore.Cluster.ReadRawDatabase(context, databaseName, out _);
+                            databaseRecord.TryGet(nameof(DatabaseRecord.Disabled), out bool dbDisabled);
```

Comment:
Check DatabasesInfoNotificationSender.IsDatabaseDisabled implementation. It takes more factors into account (e.g. DatabaseState).

Reply:
done